### PR TITLE
.github/workflows: Separate out the e2e jobs into workflows

### PR DESCRIPTION
**Pull request by timflannagan**
Update the .github/workflows/test-scripts.yml GH action workflow and
move all the jobs listed there into separate workflows. This allows us
to retest a single workflow at a time as a workaround to GH not allowing
you to retest individual jobs in a workflow.

**Description of the change:**

**Motivation for the change:**

**Reviewer Checklist**
- [ ] Implementation matches the proposed design, or proposal is updated to match implementation
- [ ] Sufficient unit test coverage
- [ ] Sufficient end-to-end test coverage
- [ ] Docs updated or added to `/doc`
- [ ] Commit messages sensible and descriptive

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+name: build
+on:
+  pull_request:
+    paths:
+    - '**'
+    - '!doc/contributors/**'
+    - '!doc/design/**'
```
Comment (timflannagan) [inline .github/workflows/build.yml:7]: This still needs to be updated to handle the periodic testing case if I'm reading the following section in (the now deleted) test-scripts.yml workflow definition:

```
  schedule:
  - cron: '0 0 * * *' # daily to pick up releases
```

/hold

**Comment by timflannagan** [comment]
@exdx This is probably a good time to revisit how we're naming these workflow/jobs instead of that other PR I posted last week. I think removing the upstream-* prefix from anything is a step in the right direction - any opinion on naming conventions, besides that?

**Comment by exdx** [comment]
> @exdx This is probably a good time to revisit how we're naming these workflow/jobs instead of that other PR I posted last week. I think removing the upstream-* prefix from anything is a step in the right direction - any opinion on naming conventions, besides that?

This looks great @timflannagan. The other thing I'd like to change is the duplicate name as I brought up in the prior PR, things like `build/build` and `unit/unit` -- could we have those under a `test/*` or `ci/*` action instead?

**Review by njhale — APPROVED** [review]
Thanks for cleaning this up.

/approve

**Review by dinhxuanvu — APPROVED** [review]
/lgtm

**Pull request merged**